File: account-jpa/src/main/java/me/tony/tonyirl/acount/jpa/repository/AccountRepository.java
package me.tony.tonyirl.acount.jpa.repository;

import me.tony.tonyirl.acount.jpa.entity.Account;
import org.springframework.data.jpa.repository.JpaRepository;

public interface AccountRepository extends JpaRepository<Account, Long> {

    Account findByAccountKeyEquals(String accountKey);

    Account findByEmailEquals(String email);

    Account findByMobileEquals(String mobile);
}

File: bootz-account-center/src/main/kotlin/cn/bootz/account/inf/jpa/repository/AccountRepository.kt
package cn.bootz.account.inf.jpa.repository

import cn.bootz.account.inf.jpa.entity.AccountBase
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor

interface AccountRepository : JpaRepository<AccountBase, Long>, JpaSpecificationExecutor<AccountBase>